display-composable-mapview/src/main/java/com/esri/arcgismaps/sample/displaycomposablemapview/MapViewWithCompose.kt

```diff
@@ -0,0 +1,52 @@
+package com.esri.arcgismaps.sample.displaycomposablemapview
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.Lifecycle
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.MapView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+/**
+ * Wraps the MapView in a Composable function.
+ */
+@Composable
+fun MapViewWithCompose(
+    lifecycle: Lifecycle,
+    arcGISMap: ArcGISMap,
+    viewpoint: Viewpoint,
+    onSingleTap: (mapPoint: Point?) -> Unit = {},
+) {
+    AndroidView(
+        // modifiers are used to set layout parameters
+        modifier = Modifier.fillMaxSize(),
+        // the factory parameter provides a context to create a classic Android view
+        // called when the composable is created, but not when it's recomposed
+        factory = { context ->
+            MapView(context).also { mapView ->
+                // add the MapView to the lifecycle observer
+                lifecycle.addObserver(mapView)
+                // set the map
+                mapView.map = arcGISMap
+                // launch a coroutine to collect map taps
+                CoroutineScope(Dispatchers.Default).launch {
```

Comment:
I think `CoroutineScope()` will note inherit a coroutine context, and there is not one to inherit here anyway, so this coroutine will essentially be an async function. It would be better to use the lifecycle owner here as well:
```suggestion
                lifecycleOwner.lifecycleScope.launch(Dispatchers.Default) {
```